components/calcs-page/ImtCalcList.tsx

```diff
@@ -0,0 +1,74 @@
+'use client';
+
+import React, { useState } from 'react';
+import { InputSkeleton } from './InputSkeleton';
+import * as yup from 'yup';
+import { ImtCalcFormula } from './ImtCaclFormula';
+
+const schema = yup.object().shape({
+  height: yup
+    .number()
+    .positive()
+    .min(100, 'Не менше 100')
+    .max(220, 'Не більше 220'),
+
+  weight: yup
+    .number()
+    .positive()
+    .min(40, 'Не менше 40')
+    .max(130, 'Не більше 130'),
+});
+
+export const ImtCalcList = () => {
+  const [weight, setWeight] = useState<number>(0);
+  const [height, setHeight] = useState<number>(0);
+  const [errors, setErrors] = useState<{ height?: string; weight?: string }>(
+    {}
+  );
+
+  const changeWeight = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setWeight(Number(e.target.value));
+  };
+  const changeHeight = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setHeight(Number(e.target.value));
+  };
+
+  const validate = async () => {
+    try {
+      await schema.validate({ height, weight }, { abortEarly: false });
+      setErrors({});
+    } catch (err) {
+      const validationErrors: Record<string, string> = {};
+      (err as yup.ValidationError).inner.forEach(error => {
+        if (error.path) {
+          validationErrors[error.path] = error.message;
+        }
+      });
+      setErrors(validationErrors);
+    }
+  };
```

Comment:
Validation is broken here. You validate all values when onBlur some of 2 objects, fix it